rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -35,17 +35,23 @@ struct AnySubscriptionCallback
   using SharedPtrCallback = std::function<void(const std::shared_ptr<MessageT> &)>;
   using SharedPtrWithInfoCallback =
       std::function<void(const std::shared_ptr<MessageT> &, const rmw_message_info_t &)>;
+  using ConstSharedPtrCallback = std::function<void(const std::shared_ptr<const MessageT> &)>;
```

Comment:
But keeping the reference for unique pointers implies that the caller as well as the callee have a shared handle. If the caller lets the variable go out of scope while the callee is still holding the reference any access will access invalid memory.